com.unity.multiplayer.mlapi/Prototyping/NetworkTransform.cs

```diff
@@ -12,282 +13,181 @@ namespace MLAPI.Prototyping
+                m_NetworkState.Value.Position = transform.position;
+                m_NetworkState.Value.Rotation = transform.rotation;
+                m_NetworkState.Value.Scale = transform.lossyScale;
             }
-        }
-
-        private Vector3 m_OldPosition;
-        private Quaternion m_OldRotation;
-        private Vector3 m_OldScale;
-
-        private NetworkVariable<Vector3>.OnValueChangedDelegate m_PositionChangedDelegate;
-        private NetworkVariable<Quaternion>.OnValueChangedDelegate m_RotationChangedDelegate;
-        private NetworkVariable<Vector3>.OnValueChangedDelegate m_ScaleChangedDelegate;
-
-        // todo really not happy with that one, hopefully we can have a cleaner solution with reparenting.
-        private void SetWorldScale(Vector3 globalScale)
-        {
-            m_Transform.localScale = Vector3.one;
-            var lossyScale = m_Transform.lossyScale;
-            m_Transform.localScale = new Vector3(globalScale.x / lossyScale.x, globalScale.y / lossyScale.y, globalScale.z / lossyScale.z);
-        }
 
-        public bool CanUpdateTransform()
-        {
-            return (IsClient && TransformAuthority == Authority.Client && IsOwner) || (IsServer && TransformAuthority == Authority.Server) || TransformAuthority == Authority.Shared;
+            m_NetworkState.SetDirty(true);
         }
 
-        private void Awake()
+        private void ApplyNetworkState(NetworkState netState)
         {
-            m_Transform = transform;
+            InLocalSpace = netState.InLocalSpace;
+            if (InLocalSpace)
+            {
+                transform.localPosition = netState.Position;
+                transform.localRotation = netState.Rotation;
+                transform.localScale = netState.Scale;
+            }
+            else
+            {
+                transform.position = netState.Position;
+                transform.rotation = netState.Rotation;
+                // transform.lossyScale = netState.Scale;
+                transform.localScale = Vector3.one;
+                var lossyScale = transform.lossyScale;
+                transform.localScale = new Vector3(netState.Scale.x / lossyScale.x, netState.Scale.y / lossyScale.y, netState.Scale.z / lossyScale.z);
+            }
         }
 
-        public override void OnNetworkSpawn()
+        private void OnNetworkStateChanged(NetworkState oldState, NetworkState newState)
         {
-            void SetupVar<T>(NetworkVariable<T> v, T initialValue, ref T oldVal)
+            if (!NetworkObject.IsSpawned)
             {
-                v.Settings.SendTickrate = FixedSendsPerSecond;
-                v.Settings.SendNetworkChannel = Channel;
-                if (CanUpdateTransform())
-                {
-                    v.Value = initialValue;
-                }
-                oldVal = initialValue;
+                // todo should never happen but yet it does! maybe revisit/dig after NetVar updates and snapshot system lands?
```

Comment:
I'd add a Jira on this, floating todos like this never get fixed in my experience